include/openenclave/internal/defs.h

```diff
@@ -8,7 +8,7 @@
 
 /* OE_WEAK_ALIAS */
 #define OE_WEAK_ALIAS(OLD, NEW) \
-    extern __typeof(OLD) NEW __attribute__((weak, alias(#OLD)))
+    extern __typeof(OLD) NEW __attribute__((__weak__, alias(#OLD)))
```

Comment:
@mikbras, @anakrish This fix prevent us from hitting a symbol collision with MUSL libc (which uses 'weak' as a macro def identical to `OE_WEAK_ALIAS`) but I'm not sure where `__weak__` is coming from. All the GCC docs use it in the default form which we were also using.

@aserdean MUSL also updated `alias` to `__alias__` in their definition, not sure if it matters to us to also update that; I haven't seen a collision of the `alias` symbol itself with anything else they define.

Reply:
@CodeMonkeyLeet : The attribute syntax optionally allows to surround attribute names
with "__" in order to avoid collisions with macros of the same name
(see https://gcc.gnu.org/onlinedocs/gcc/Attribute-Syntax.html).

Reply:
@aserdean Thanks! That was helpful.